test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/individual/BusinessNameConfirmationControllerSpec.scala

```diff
@@ -102,11 +163,31 @@ class BusinessNameConfirmationControllerSpec extends ControllerBaseSpec
     }
   }
 
-  "submit" when {
+  "show" when {
     "there is no name in the users session" should {
       "redirect the user to the business name page" in new Setup {
         mockAuthSuccess()
+        mockFetchAllBusinesses(Right(Seq()))
+        val response: Future[Result] = controller.show(id)(fakeRequest)
+
+        status(response) mustBe SEE_OTHER
+        redirectLocation(response) mustBe Some(controllers.routes.BusinessNameController.show(id).url)
+      }
+    }
+    "there is a name in the users session" must {
+      "return OK with the page contents" in new Setup {
+        mockAuthSuccess()
+
 
+      }
+    }
+  }
```

Comment:
I would prefer to combine this with the above show tests.
Shouldn't be too hard with a bit of rearranging of the wording:
"show" must "redirect the user to the business name page" ...